### ISSUE TITLE
[web] Exception during scroll/drag (e.g. in ListView)

### ISSUE THREAD

**Issue by a-marenkov** [flutter/flutter#49995]
If scrollable content dragged and cursor of mouse goes out of window bounds (without release of the button) following exception is thrown:
```
errors.dart:147 Uncaught Error: Assertion failed: org-dartlang-sdk:///flutter_web_sdk/lib/_engine/engine/pointer_converter.dart:271:18
!state.down
is not true
    at Object.throw_ [as throw] (errors.dart:196)
    at Object.assertFailed (errors.dart:26)
    at _engine.PointerDataConverter.new.convert (pointer_converter.dart:271)
    at _engine._PointerAdapter.new.[_convertEventsToPointerData] (pointer_binding.dart:465)
    at pointer_binding.dart:411
    at pointer_binding.dart:390
    at HTMLElement.<anonymous> (pointer_binding.dart:195)
```

**After the exception is thrown, application stops reacting to clicks (not reacting to clicks on buttons or attempts to drag)**

## Steps to Reproduce
It's easier to reproduce if window of a browser is not maximized to fullscreen.

```dart
class MyHomePage extends StatelessWidget {
  const MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  Widget build(BuildContext context) {
    return Container(
      color: Colors.blueGrey,
      child: ListView.builder(
        itemCount: 100,
        itemBuilder: (context, index) {
          return Container(
            width: double.infinity,
            height: 128.0,
            alignment: Alignment.center,
            child: RaisedButton(
              child: Text('Press me'),
              onPressed: () {
                print('pressed $index at ${DateTime.now().toString()}');
              },
            ),
          );
        },
      ),
    );
  }
}
```

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel beta, v1.13.6, on Linux, locale en_US.UTF-8) 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.0)
[✓] Chrome - develop for the web
[✓] Android Studio (version 3.5)
[✓] IntelliJ IDEA Community Edition (version 2019.2)
[✓] Connected device (3 available)
• No issues found!

**Comment by VladyslavBondarenko** [comment]
Hi @a-marenkov 
It was not ListView issue specifically, fixed at https://github.com/flutter/flutter/issues/47440
You can switch to master channel or to wait when beta will be updated
I'm closing the current one as duplicate.
If you disagree please write in the comments and I will reopen it.
Thank you

**Comment by a-marenkov** [comment]
@VladyslavBondarenko

**Thanks!** 

I checked it on master and it's working perfectly regarding the issue!



Unfortunately, i can't use master for production, since it has other issues.
I'm not going to file the issue (since it's on master), but just FYI - 
I have dialog with gradient background and it fails when i call `showDialog` due to:

Assertion failed: org-dartlang-sdk:///flutter_web_sdk/lib/_engine/engine/dom_canvas.dart:74:12
paint.shader == null
is not true

**Comment by VladyslavBondarenko** [comment]
@a-marenkov 
I tried, but couldn't reproduce the mentioned error on master 1.14.7-pre.83

<details>
 <summary>my code sample</summary>

```
import 'package:flutter/material.dart';

void main() => runApp(MaterialApp(home: MyHomePage()));

class MyHomePage extends StatelessWidget {

  @override
  Widget build(BuildContext context) {
    return Center(
      child: RaisedButton(
        child: Text('Press me'),
        onPressed: () {
          _showDialog(context);
        },
      ),
    );
  }

  void _showDialog(context) {
    showDialog(
      context: context,
      builder: (BuildContext context) {
        return Container(
            decoration: BoxDecoration(
              gradient: LinearGradient(
                colors: [Colors.red, Colors.grey],
              ),
            ),
          child: AlertDialog(
            title: Text("Alert Dialog title"),
            content: Container(
              padding: const EdgeInsets.all(20.0),
              decoration: BoxDecoration(
                gradient: LinearGradient(
                  colors: [Colors.red, Colors.grey],
                ),
              ),
              child: Text("Alert Dialog body"),
            ),
            actions: <Widget>[
              FlatButton(
                child: Text("Close"),
                onPressed: () {
                  Navigator.of(context).pop();
                },
              ),
            ],
          ),
        );
      },
    );
  }
}
```

</details>

**Comment by a-marenkov** [comment]
Sorry. I ran my code and didn't really dig into the issue. 

Turns out the problem is not with the gradient in dialog.

Sending the sample code to reproduce the issue.

SingleChildScrollView causes the issue in this case (scroll added to make content available for small screen sizes, website can be opened on such mobile phones).

Removing SingleChildScrollView or moving it up the Widget tree "solves" the issue, but content either wouldn't scroll or look differently.
So it does not really help me, maybe i should think of different implementation.

Similar code works on beta.

```dart
class MyHomePage extends StatelessWidget {
  const MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  Widget build(BuildContext context) {
    return Center(
      child: RaisedButton(
        child: Text('Press me'),
        onPressed: () {
          _showDialog(context);
        },
      ),
    );
  }

  void _showDialog(context) {
    showDialog(
      context: context,
      builder: (BuildContext context) {
        return Center(
          child: Container(
            constraints: BoxConstraints(maxWidth: 450.0, maxHeight: 600.0),
            margin: const EdgeInsets.all(24.0),
            child: ClipRRect(
              borderRadius: BorderRadius.all(Radius.circular(16.0)),
              clipBehavior: Clip.antiAlias,
              child: SingleChildScrollView(
                child: Container(
                  decoration: BoxDecoration(
                    gradient: const LinearGradient(
                      begin: Alignment(-0.05, -1.0),
                      end: Alignment(0.05, 1.0),
                      stops: [0.66, 1.0],
                      colors: [
                        Color(0xFFcfdfd2),
                        Color(0xFF042a85),
                      ],
                    ),
                    borderRadius: BorderRadius.all(Radius.circular(16.0)),
                  ),
                  child: SizedBox(width: double.infinity, height: 600.0),
                ),
              ),
            ),
          ),
        );
      },
    );
  }
}
```

**Comment by VladyslavBondarenko** [comment]
@a-marenkov 
Yes, I see what do you mean, thank you
You can create separate issue for this

**Comment by vidklopcic** [comment]
This issue was resolved (at least on commit `fdac8a9d1f321388d71e35f825b40c05901e74f0`), but it started happening again (I'm currently on `dc80d499f3355874ea0fd1871f7a3012a7b7b9d4`).
